doc/reference/reference_lua/checks.rst

```diff
@@ -0,0 +1,516 @@
+..  checks-module:
+
+Module checks
+=============
+
+**Since:** :doc:`2.11.0 </release/2.11.0>`
+
+The ``checks`` module provides the ability to check the types of arguments passed to a Lua function.
+You need to call the :ref:`checks(type_1, ...) <checks-checks>` function inside the target Lua function and pass :ref:`one or more <checks_number_of_arguments>` type qualifiers to check the corresponding argument types.
+There are two types of type qualifiers:
+
+-   A :ref:`string type qualifier <checks_string_type_qualifier>` checks whether a function's argument conforms to the specified type.
+-   A :ref:`table type qualifier <checks_table_type_qualifiers>` checks whether the values of a table passed as an argument conform to the specified types.
+
+.. NOTE::
+
+    For earlier versions, you can install the ``checks`` module from the `Tarantool rocks repository <https://www.tarantool.io/en/download/rocks>`_.
+
+.. _checks_loading:
+
+Loading checks
+--------------
+
+The :ref:`checks API <checks_api_reference>` is available in a script without loading the module using ``require('checks')``.
+
+
+.. _checks_number_of_arguments:
+
+Number of arguments to check
+----------------------------
+
+Depending on the number of arguments to be checked, you can pass one or more type qualifiers to the :ref:`checks(type_1, ...) <checks-checks>` function.
+
+.. _checks_one_argument:
+
+One argument
+~~~~~~~~~~~~
+
+In the example below, the ``checks`` function accepts a ``string`` type qualifier to verify that only a string value can be passed to the ``greet`` function.
+Otherwise, an error is raised.
+
+..  literalinclude:: /code_snippets/test/checks/checks_string_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+.. _checks_multiple_arguments:
+
+Multiple arguments
+~~~~~~~~~~~~~~~~~~
+
+To check the types of several arguments, you need to pass the corresponding type qualifiers to the ``checks`` function.
+In the example below, both arguments should be string values.
+
+..  literalinclude:: /code_snippets/test/checks/checks_string_multiple_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+To turn off checking of specific arguments, use the :ref:`? placeholder <checks_any_type>`.
```

Comment:
Do I get it right that `?` is for checking arbitrary subsets of arguments? For example, in `f(a, b, c, d)`, check only `a` and `d`? If yes, then maybe show an example?

Reply:
Also, `skip checking` seems better to me here than `turn off checking`.

Reply:
> Do I get it right that `?` is for checking arbitrary subsets of arguments? For example, in `f(a, b, c, d)`, check only `a` and `d`? If yes, then maybe show an example?

Input count must match with checks expected count (excluding varargs).

```lua
tarantool> g = function(a, b) checks('number', 'number') end
---
...

tarantool> g(1, 2)
---
...

tarantool> g(1)
---
- error: '[string "g = function(a, b) checks(''number'', ''number'')..."]:1: bad argument
    #2 to nil (number expected, got nil)'
... 

tarantool> g = function(a, b) checks('number') end
---
...

tarantool> g(1, 2)
---
- error: '[string "g = function(a, b) checks(''number'') end"]:1: checks: argument
    "b" is not checked'
...

tarantool> g = function(a, b) checks('number', '?') end
---
...

tarantool> g(1)
---
...

```

So, in case you want to check some args and you do not care about other args, you must check them with `'?'` since you cannot simply skip them. It makes even more sence in case you don't want to check some argument in the middle. It is similar for table fields -- if you want to allow optional field X with arbitrary value, you use `'?'`.

Reply:
> Do I get it right that ? is for checking arbitrary subsets of arguments? For example, in f(a, b, c, d), check only a and d? If yes, then maybe show an example?

You are right. Will update the example to make it more clear.